packages/nextjs/app/builders/page.tsx

```diff
@@ -0,0 +1,95 @@
+"use client";
+
+import { useState } from "react";
+import type { NextPage } from "next";
+import { FaExternalLinkAlt } from "react-icons/fa";
+import { useScaffoldReadContract } from "~~/hooks/scaffold-eth";
+
+type Builder = {
+  name: string;
+  address: `0x${string}`;
+};
+
+const Home: NextPage = () => {
+  const [builders] = useState([
+    {
+      name: "Abdulyekeen Lukman",
+      address: "0x186a761645f2A264ad0A655Fb632Ca99150803A9",
+    },
+    {
+      name: "Favvie Kenpachi",
+      address: "0x21Be2291f91EA2A1d1EB65DbBea2dA8886Ad7a3E",
+    },
+    {
+      name: "Bello Abraham",
+      address: "0x28482B1279E442f49eE76351801232D58f341CB9",
+    },
+    {
+      name: "Samson Aderonmu",
+      address: "0x62CeF3Ca8b52a9C69a17236CA2c56Cdb7a383E8e",
+    },
+    {
+      name: "Michael Ojekunle",
+      address: "0x7429CbD5eD20736645723E972bE60B7F6BF5959c",
+    },
+  ] as Builder[]);
```

Comment:
We’d prefer to avoid adding builders manually. Could you think of a way to automatically identify the builders, such as those who have checked in and then who have created a personal page?

Reply:
@phipsae Is there a way for me to fetch the name of a builder? The `CheckedIn` event (through which I fetched the builders) only contains their address.

Reply:
Hi @pbkompasz,
I believe we can simplify the page by omitting the builder's name. Displaying the ENS name or address should be sufficient. Let me know what you think!

Reply:
What about having the best of both worlds? 

- Pull checked-in addresses dynamically from the smart contract events
- Pull builder pages dynamically from the routes available in the repo. If a checked-in address has a builder page, list it next to their name. (I also recommend sorting the list at the end to put builder who have pages on top, as a little bonus for their work and better UX for someone who wants to see the pages)
- Use the address component in scaffold-eth to display their ENS name instead of their 0x... if they have one
- (Before) store the names we do know in a "constants.ts" sort of enum/dict struct. If the addresses from checked-in events match one of these known address/name pairs, display the name. If not, fallback to EVM/ENS. Or have name as a third column, next to EVM/ENS (if no name, leave blank)

This is a common data processing pattern in web3, where your sources of data are coming from different places and don't necessarily fully overlap with each other. Having conditional states in your frontend components ready to handle that is very professional 👌 

Like `(if isCheckedIn && hasBuilderPage && isInKnownNames)` kind of thing

Reply:
I went with @derrekcoleman's idea. There are now two columns: the first displays the address or ENS using the `SE2/Address` component, and the second contains the builder's name (if available) and a link to their profile(if available).
The list is sorted to show builders with a profile page first.